zircon.core/src/commonMain/kotlin/org/hexworks/zircon/internal/fragment/impl/table/TableColumn.kt

```diff
@@ -0,0 +1,42 @@
+package org.hexworks.zircon.internal.fragment.impl.table
+
+import org.hexworks.zircon.api.Components
+import org.hexworks.zircon.api.component.Component
+
+/**
+ * This class represents the definition of a column in a table. It provides means to get the value
+ * from a model object and wrap that value into a [Component]. This component is then displayed
+ * as table cell.
+ *
+ * @param M the type of the model. In other words: Every element/row in the table has this type
+ * @param V the type of the value of each cell in this column
+ * @param C type of the [Component] used to represent each cell
+ */
+open class TableColumn<M : Any, V : Any, C : Component>(
```

Comment:
Does this need to be `open`?

Reply:
I wanted to give the users of the API as much flexibility as possible. You should not be restricted to only use the predefined `TableColumns` or this constructor. Maybe you have your own fancy implementation and want to use it over and over again so it thought it should be possible to extend this class.